src/handlers.ts

```diff
@@ -304,6 +304,101 @@ export async function replyAction(
     });
 }
 
+export function createUnfollowActionHandler(accountService: AccountService) {
+    return async function unfollowAction(
+        ctx: Context<{ Variables: HonoContextVariables }>,
+    ) {
+        const handle = ctx.req.param('handle');
+        const apCtx = fedify.createContext(ctx.req.raw as Request, {
+            db: ctx.get('db'),
+            globaldb: ctx.get('globaldb'),
+            logger: ctx.get('logger'),
+        });
+
+        const actorToUnfollow = await lookupObject(apCtx, handle);
+
+        if (!isActor(actorToUnfollow)) {
+            return new Response(null, {
+                status: 404,
+            });
+        }
+
+        const account = await accountService.getDefaultAccountForSite(
+            ctx.get('site'),
+        );
+
+        if (actorToUnfollow.id!.href === account.ap_id) {
+            return new Response(null, {
+                status: 400,
+            });
+        }
+
+        let unfolloweeAccount = await accountService.getAccountByApId(
+            actorToUnfollow.id!.href,
+        );
+
+        // TODO I think we can exit early here - there is obviously no follow relation if there is no account
+        if (!unfolloweeAccount) {
+            unfolloweeAccount = await accountService.createExternalAccount(
+                await mapActorToExternalAccountData(actorToUnfollow),
+            );
+        }
```

Comment:
Yeh i think we can bail early here - If we can't find an account for the actor being unfollowed, then there isn't a record of them following us so no need to carry on

Reply:
Yess, I was thinking the same!